src/include/daos_obj.h

```diff
@@ -127,6 +127,12 @@ enum {
 	DAOS_COND_AKEY_FETCH	= (1 << 6),
 	/** Mask for convenience */
 	DAOS_COND_MASK		= ((1 << IO_FLAGS_COND_BITS) - 1),
+
+	/**
+	 * Not copy application buffer when cache modification on client
+	 * for DTX.
+	 */
```

Comment:
One comment about this is that this does not get exposed through to the array API (src/include/daos_array.h).
maybe we should make it a transaction property? like daos_tx_set_prop(th, flag); ?
maybe we should discuss more on how to set this.

Reply:
Can we export the API through array API, such as:
daos_array_write(..., flags);

Because the DAOS_ZERO_COPY is per modification based flag. It allow the application to decide whether zero copy based on current modification. Means different modifications within the same TX can have different behavior. If we control that via daos_tx_set_prop(th, flag), that is almost equal to per TX based, seems not flexible.

Reply:
Another benefit of extend array API is that we can support conditional array modification.